Lib/pdb.py

```diff
@@ -1375,13 +1447,15 @@ def do_source(self, arg):
 
     complete_source = _complete_expression
 
-    def _print_lines(self, lines, start, breaks=(), frame=None):
+    def _print_lines(self, lines, start, breaks=(), frame=None, instructions=None):
         """Print a range of lines."""
         if frame:
             current_lineno = frame.f_lineno
             exc_lineno = self.tb_lineno.get(frame, -1)
         else:
             current_lineno = exc_lineno = -1
+        if instructions:
+            inst = next(instructions)
```

Comment:
I suppose the whole iteration will become a simple `for instr in instructions` if we pass a list of instructions to `_print_lines` rather than an iterator. Is there any reason for passing an iterator?

Reply:
Not really. We have a for loop for the lines and we need to pick the instructions that belong to this line. So we can never use a simple for loop. Another not so critical reason is that it saves some time/space because we might not exhaust the interator in `li` case.

Reply:
Indeed, I misunderstood it at first glance. Maybe it is worth adding a comment explaining the whole line-instructions iteration